processing/src/main/java/io/druid/query/groupby/epinephelinae/GroupByQueryEngineV2.java

```diff
@@ -219,7 +219,7 @@ private static boolean isArrayAggregateApplicable(
       final AggregatorFactory[] aggregatorFactories = query
           .getAggregatorSpecs()
           .toArray(new AggregatorFactory[query.getAggregatorSpecs().size()]);
-      final int requiredBufferCapacity = BufferArrayGrouper.requiredBufferCapacity(
+      final long requiredBufferCapacity = BufferArrayGrouper.requiredBufferCapacity(
```

Comment:
`BufferArrayGrouper.requiredBufferCapacity` must also be modified to return a `long`.

Also, please add a regression test for this.

And thank you for finding the bug!